discovery-infra/test_infra/junit_report/junit_test_suite.py

```diff
@@ -0,0 +1,54 @@
+import os
+from typing import Callable, Union, Type
+
+import decorator
+from junit_xml import TestSuite, to_xml_report_string
+
+from .junit_test_case import JunitTestCase
+
+
+class JunitTestSuite:
+    _report_dir: str
+    _func: Union[Callable, None]
+    _suite: Union[TestSuite, None]
+    _name = str
+
+    def __init__(self, report_dir="./"):
+        self._report_dir = report_dir
+        self._func = None
+        self._suite = None
+        self._name = None
+
+    def __call__(self, function: Callable):
+        self._func = function
+        self._name = function.__name__
+
+        def wrapper(_, obj: Type[object], *args, **kwargs):
+            try:
+                value = function(obj, *args, **kwargs)
+            except BaseException as e:
+                raise e
+            finally:
+                self._collect(obj.__class__)
+            return value
+
+        return decorator.decorator(wrapper, function)
+
+    def _collect(self, klass: Type[object]):
+        self._suite = TestSuite(name=f"{klass.__name__}_{self._name}")
+
+        for maybe_case in klass.__dict__.values():
+            if hasattr(maybe_case, "case") and isinstance(maybe_case.case, JunitTestCase):
```

Comment:
Maybe it should be guarded also with `inspect.isfunction(v)`

Reply:
If it has case attribute and its instance of JunitTestCase it's doesn't  matter if its function